apps/portal/controllers/apis/analytics.jag

```diff
@@ -58,269 +58,277 @@
     var AnalyticsWebServiceConnector = Packages.org.wso2.carbon.analytics.jsservice.AnalyticsWebServiceConnector;
     var AnalyticsWebServiceStub = Packages.org.wso2.carbon.analytics.webservice.stub.AnalyticsWebServiceStub;
     var AuthenticationAdminStub = Packages.org.wso2.carbon.authenticator.stub.AuthenticationAdminStub;
-    var analyticsWSUrl = carbon.server.address(HTTPS_TRANSPORT) + ANALYTICS_SERVICE;
-    var authenticationWSUrl = carbon.server.address(HTTPS_TRANSPORT) + AUTHENTICATION_SERVICE;
-    var analyticsStub = new AnalyticsWebServiceStub(analyticsWSUrl);
-    var authenticationStub = null;
-    var serviceClient = analyticsStub._getServiceClient();
-    var options = serviceClient.getOptions();
-    options.setManageSession(true);
-    response.contentType = CONTENT_TYPE_JSON;
 
-    var authParam = request.getHeader(AUTHORIZATION_HEADER);
-    if (authParam != null) {
-        credentials = JSUtils.authenticate(authParam);
-        authenticationAdminStub = new AuthenticationAdminStub(authenticationWSUrl);
-        authenticationAdminStub.login(credentials[0], credentials[1], LOCALHOST);
-        var serviceContext = authenticationAdminStub._getServiceClient().getLastOperationContext()
-                .getServiceContext();
-        var sessionCookie = serviceContext.getProperty(HTTPConstants.COOKIE_STRING);
-        options.setProperty(HTTPConstants.COOKIE_STRING, sessionCookie);
-    } else {
-        var token = session.get(AUTH_TOKEN);
-        if (token != null) {
-            options.setProperty(HTTPConstants.COOKIE_STRING, token);
+
+    if (!AnalyticsWebServiceStub && !AuthenticationAdminStub) {
+
+        var analyticsWSUrl = carbon.server.address(HTTPS_TRANSPORT) + ANALYTICS_SERVICE;
+        var authenticationWSUrl = carbon.server.address(HTTPS_TRANSPORT) + AUTHENTICATION_SERVICE;
+        var analyticsStub = new AnalyticsWebServiceStub(analyticsWSUrl);
+        var authenticationStub = null;
```

Comment:
@lasanthafdo 
I checked the JS i guess we can remove authenticationStub, I will fix and do a testing and update the pull request